starter-code/src/clue.js

```diff
@@ -85,10 +87,138 @@ name: Patio
+  ocupation:   'Scientist'
+}
+
+let profPlum = {
+  first_name:   'Victor',
+  last_name:    'Plum',
+  color:        'purple',
+  description:  'Billionare video game designer',
+  age:          22,
+  image:        'https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg',
+  occupation:   'Designer'
+}
+
+let missScarlet = {
+
+  first_name:   'Kasandra',
+  last_name:    'Scarlet',
+  color:        'red',
+  description:  'She is an A-list movie star with a dark past',
+  age:          31,
+  image:        'https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg',
+  occupation:   'Actor'
+}
+
+let mrsPeacock = {
+
+  first_name:   'Eleanor',
+  last_name:    'Peacock',
+  color:        'blue',
+  description:  'She is from a wealthy family and uses her status and money to earn popularity',
+  age:          36,
+  image:        'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+  occupation:   'Socialité'
+}
+
+let mrMustard = {
+
+  first_name:   'Jack',
+  last_name:    'Mustard',
+  color:        'yellow',
+  description:  'He is a former football player who tries to get by on his former glory',
+  age:          62,
+  image:        'https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg',
+  occupation:   'Retired Football player'
+}
+
+charactersArray.push(mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard)
 
 // Rooms' Collection
-var roomsArray = [];
+let roomsArray = ["Dinning Room","Conservatory", "Kitchen", "Study", "Library","Billiard Room", "Lounge", "Ballroom", "Hall","Spa", "Living Room", "Observatory", "Theater", "Guest House", "Patio"];
 
 // Weapons Collection
-var weaponsArray = [];
+
+const weaponsArray = [
+  {name: "rope", weight: 10},
+  {name: "knife", weight: 8},
+  {name: "knife" ,weight: 8},
+  {name: "candlestick", weight: 2},
+  {name: "dumbbell", weight: 30},
+  {name: "poison", weight: 2},
+  {name: "axe", weight: 15},
+  {name: "bat", weight: 13},
+  {name: "trophy", weight: 25},
+  {name: "pistol", weight: 20}
+]
+
+function randomizer(arr){
+  let len = arr.length
+  let rnd = Math.floor(Math.random() * len) 
+  return arr[rnd]
+}
+
+// RANDOM SELECTOR
+function randomSelector(arr){
+  let len = arr.length
+  if (len == 0) {
+    return undefined
+  }
+  else { 
+    randomizer(arr)
+    return randomizer(arr)
+  }
+
+}
+const misteryRoom = randomSelector(roomsArray)
+const misteryWeapon = randomSelector(weaponsArray)
+const misteryKiller = randomSelector(charactersArray)
+const misteryReveal =pickMistery()
+
+// Pick Mistery
+function pickMistery(){
+  let misteryEnvelope = []
+  misteryEnvelope.push(misteryKiller)
+  misteryEnvelope.push(misteryWeapon)
+  misteryEnvelope.push(misteryRoom)
+  return misteryEnvelope
+}
+
+pickMistery(charactersArray)
+
+function revealMistery(array){
+  const firstName = array[0].first_name
+  const lastName = array[0].last_name
+  const weapon = array[1].name
+  const place = array[2].name
```

Comment:
El código de ustedes funciona con los parámetros de Jasmine, pero no con el arreglo `roomsArray `que ustedes crearon. Ahorita me regresa `Doctor Orchid killed Mr.Boddy using the candlestick in the undefined!!!!`, justo porque está buscando la propiedad `name` en la cadena "Kitchen" (por ejemplo). Para que eso funcionara, necesitarían que roomsArray fuera un arreglo de objetos.